### PR TITLE
BTFS-975: UnixFS metadata support for Trickle format

### PR DESCRIPTION
**Pull request by steveyeom**
1. Added UnixFS metadata support for trickle format.
2. Added unit tests in trickle_test.go
3. Passed all the unit tests in UnixFS.

### CODE REVIEW

**Comment by steveyeom** [comment]
I will use this PR for BTFS-887: the original metadata support ticket for trickle format.

File: importer/helpers/dagbuilder.go [hunk 1]
```diff
@@ -33,6 +33,8 @@ type DagBuilderHelperInterface interface {
 	Done() bool
 	Add(ipld.Node) error
 	NewLeafDataNode(pb.Data_DataType) (ipld.Node, uint64, error)
+    FillNodeLayer(*FSNodeOverDag, pb.Data_DataType) error
```
Comment (taiyangc) [inline importer/helpers/dagbuilder.go]: go fmt

File: importer/trickle/trickle_test.go [hunk 2]
```diff
@@ -665,3 +665,190 @@ func TestAppendSingleBytesToEmpty(t *testing.T) {
...
+
+	mdb := db.GetMetaDb()
+	nd := mdb.GetMetaDagRoot()
+
+	pbnd, ok := nd.(*merkledag.ProtoNode)
+	if !ok {
+		return nil, merkledag.ErrNotProtobuf
+	}
+
+	//return pbnd, nil
```
Comment (taiyangc) [inline importer/trickle/trickle_test.go:695]: delete?

**Review by taiyangc — COMMENTED** [review]
code is good, just needs a go fmt

**Review by taiyangc — APPROVED** [review]

**Pull request merged**